### PR TITLE
DS offers repo refactoring - Part 1

### PR DESCRIPTION
**Pull request by dstarr**
Addresses, but does not close, #379 #375 

This is a very simple change and the first of several which will be required on the OffersController, but I wanted to send this in as it has a new test project and does remove at least one repo from the controller.

- Added very simple unit test project to test the offers service
- Added new contructior to IOffersRepository to take SaaSKitContext
- Put the IOffersRepo under the OfferServices class
- Removed IOffersRepo dependency from the OfferController class
- Refactored controller
- Deleted all commented out code in the OffersDetail view
- Fully tested the landing page
- Made no changes to the DataAccess project

### CODE REVIEW

File: src/AdminSite/Startup.cs [hunk 1]
```diff
@@ -139,6 +139,8 @@ public void ConfigureServices(IServiceCollection services)
         services.Configure<CookieTempDataProviderOptions>(options => {
             options.Cookie.IsEssential = true;
         });
+
```
Comment (santhoshb-msft) [inline src/AdminSite/Startup.cs:142]: Can we add this to the InitializeRepositoryServices method, @dstarr ?

Comment (dstarr) [reply src/AdminSite/Startup.cs:142]: I don't see that as a repository. InitializeRepositoryServices is a private method that just wraps registering repos with the DI system.

Comment (santhoshb-msft) [reply src/AdminSite/Startup.cs:142]: I am thinking if its reading like InitializeRepositorys and InitializeServices ?and having them at one place would help with whats used for DI at one place.

Comment (dstarr) [reply src/AdminSite/Startup.cs:142]: I guess I don't understand as `Startup.ConfigureServices` is the containing function for registering services. The other methods you are referring to are private methods provided for clarity and brevity.

I may be misunderstanding, in which case we should just have a brief conversation.

**Review by santhoshb-msft — APPROVED** [review]
Looks good, @dstarr . Approved.

**Pull request merged**